Library/Homebrew/cmd/pull.rb

```diff
@@ -126,42 +120,58 @@ def pull
         end
       end
 
+      orig_message = message = `git log HEAD^.. --format=%B`
       if issue && !ARGV.include?("--clean")
         ohai "Patch closes issue ##{issue}"
-        message = `git log HEAD^.. --format=%B`
+        # If this is a pull request, append a close message.
+        unless message.include? "Closes ##{issue}."
+          message += "\nCloses ##{issue}."
+        end
+      end
 
-        if ARGV.include? "--bump"
-          odie "Can only bump one changed formula" unless changed_formulae.length == 1
-          formula = changed_formulae.first
-          subject = "#{formula.name} #{formula.version}"
+      if changed_formulae.empty?
+        odie "cannot bump: no changed formulae found after applying patch" if do_bump
+        is_bumpable = false
+      end
+      if is_bumpable && !ARGV.include?("--clean")
+        formula = changed_formulae.first
+        new_versions = {
+            stable: formula.stable.nil? ? nil : formula.stable.version.to_s,
+            devel: formula.devel.nil? ? nil : formula.devel.version.to_s
+        }
+        orig_subject = message.empty? ? "" : message.lines.first.chomp
+        subject = subject_for_bump(formula, old_versions, new_versions)
+        if do_bump
+          odie "No version changes found for #{formula.name}" if subject.nil?
```

Comment:
I think the rest of this `if` branch should be a no-op if `subject == orig_subject`. No need to fix something that's not broken.